contracts/src/systems/season/contracts.cairo

```diff
@@ -0,0 +1,122 @@
+use dojo::model::ModelStorage;
+use dojo::world::WorldStorage;
+use dojo::world::{IWorldDispatcher, IWorldDispatcherTrait};
+use s0_eternum::alias::ID;
+
+#[starknet::interface]
+trait ISeasonSystems<T> {
+    fn register_to_leaderboard(
+        ref self: T, hyperstructures_contributed_to: Span<ID>, hyperstructure_shareholder_epochs: Span<(ID, u16)>
+    );
+    fn claim_leaderboard_rewards(ref self: T, token: starknet::ContractAddress);
+}
+
+#[dojo::contract]
+mod season_systems {
+    use core::array::ArrayIndex;
+    use dojo::event::EventStorage;
+    use dojo::model::ModelStorage;
+    use dojo::world::WorldStorage;
+    use dojo::world::{IWorldDispatcher, IWorldDispatcherTrait};
+
+    use s0_eternum::{
+        constants::{HYPERSTRUCTURE_CONFIG_ID, WORLD_CONFIG_ID, DEFAULT_NS, ResourceTypes}, alias::ID,
+        models::{
+            config::{HyperstructureConfig, ResourceBridgeFeeSplitConfig, ResourceBridgeWhitelistConfig},
+            season::{Leaderboard, LeaderboardEntryCustomImpl, LeaderboardEntry, LeaderboardRewardClaimed}
+        },
+        systems::{
+            hyperstructure::contracts::hyperstructure_systems::InternalHyperstructureSystemsImpl,
+            config::contracts::config_systems::assert_caller_is_admin,
+            resources::contracts::resource_bridge_systems::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait}
+        },
+    };
+    use starknet::ContractAddress;
+
+    pub const SCALING_FACTOR: u256 = 1_000_000;
+
+    #[abi(embed_v0)]
+    impl SeasonSystemsImpl of super::ISeasonSystems<ContractState> {
+        fn register_to_leaderboard(
```

Comment:
check if registration period has started

Reply:
This should fail anyways as if the season isn't over the leaderboard wouldn't have been set before so `registration_end_timestamp` would be 0 so this line would panic 
```
            assert!(
                leaderboard.registration_end_timestamp > starknet::get_block_timestamp(), "Registration period is over"
            );
```
do you think it should be more explicit?

Reply:
yeah just created a test for it